app/controllers/view_components_controller.rb

```diff
@@ -4,4 +4,18 @@
 
 class ViewComponentsController < Rails::ApplicationController # :nodoc:
   include ViewComponent::PreviewActions
+
+  def system_test_entrypoint
+    if Rails.env.production?
+      render body: "Unauthorized", status: 403
+    else
+      render file: "./tmp/view_component_integrations/#{file_param[:file]}", status: 200
+    end
+  end
```

Comment:
I think we can be even more strict here, only turning on this route in test:

```suggestion
  if Rails.env.test?
    def system_test_entrypoint
      render file: "./tmp/view_component_integrations/#{file_param[:file]}", status: 200
    end
  end
```

Reply:
Good idea!